### PR TITLE
feat(s3): add Intelligent-Tiering configuration CRUD

### PR DESCRIPTION
**Pull request by ChiaraFor96**
## Summary

Adds AWS S3 Intelligent-Tiering configuration management (put, get, list, delete) through the real REST XML `?intelligent-tiering` subresource, mirroring the existing metrics configuration pattern.

## Type of change

- [ ] Bug fix (`fix:`)
- [x] New feature (`feat:`)
- [ ] Breaking change (`feat!:` or `fix!:`)
- [ ] Docs / chore

## AWS Compatibility

S3 Intelligent-Tiering is a REST XML service. The implementation follows the real AWS wire protocol:

- `PutBucketIntelligentTieringConfiguration` (PUT, 204)
- `GetBucketIntelligentTieringConfiguration` (GET by id)
- `ListBucketIntelligentTieringConfigurations` (GET without id)
- `DeleteBucketIntelligentTieringConfiguration` (DELETE, 204)

`Days` validation matches the AWS `Tiering` schema: minimum 90 for `ARCHIVE_ACCESS`, minimum 180 for `DEEP_ARCHIVE_ACCESS`, capped at 730 for both. Out-of-range values return `InvalidArgument` with `ArgumentName` and `ArgumentValue` in the XML error body, matching the real S3 error shape.

Verified with:
- `JAVA_HOME=/Users/chiara.forresi/Library/Java/JavaVirtualMachines/openjdk-25/Contents/Home ./mvnw -Dtest=S3IntelligentTieringConfigurationTest,S3IntelligentTieringConfigurationIntegrationTest test`
- 33 tests passed

## Checklist

- [x] `./mvnw test` passes locally
- [x] New or updated integration test added
- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/)

### CODE REVIEW

File: src/main/java/io/github/hectorvent/floci/services/s3/S3IntelligentTieringConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        }
+        Map<String, String> tiering = entries.get(index);
+        String accessTier = tiering.get("AccessTier");
+        String days = tiering.get("Days");
+        if (accessTier == null || !ACCESS_TIERS.contains(accessTier)) {
+            throw malformed();
+        }
+        if (days == null || !days.matches("\\d+") || Integer.parseInt(days) < 1) {
+            throw malformed();
+        }
```
[the remark this answers is not in the document]
Comment (ChiaraFor96) [reply src/main/java/io/github/hectorvent/floci/services/s3/S3IntelligentTieringConfiguration.java:177]: fixed

File: src/main/java/io/github/hectorvent/floci/services/s3/S3Controller.java [hunk 2]
```diff
@@ -333,6 +333,11 @@ public Response createBucket(@PathParam("bucket") String bucket,
             if (hasQueryParam(uriInfo, "metrics")) {
                 return handlePutBucketMetricsConfiguration(bucket, uriInfo, body);
             }
+            // Same fall-through hazard as metrics: an intelligent-tiering PUT must not become a
+            // CreateBucket.
+            if (hasQueryParam(uriInfo, "intelligent-tiering")) {
+                return handlePutBucketIntelligentTieringConfiguration(bucket, uriInfo, body);
```
[the remark this answers is not in the document]
Comment (ChiaraFor96) [reply src/main/java/io/github/hectorvent/floci/services/s3/S3Controller.java:342]: solved with the last commit

**Review by pgermosen — COMMENTED** [review]
Nicely mirrors the existing metrics-configuration pattern, checked the request-shape validation (Filter exclusivity, AccessTier values, the concurrency handling for the delete/recreate race) against the real docs and it all matches. Good to see a genuine XXE test in there too, not just a happy-path check.

One real gap though: Days is only validated as a positive integer, but AWS's own Tiering reference is explicit that the minimum is tier-specific, 90 days for ARCHIVE_ACCESS, 180 for DEEP_ARCHIVE_ACCESS, capped at 730 for both. Right now Days=1 on ARCHIVE_ACCESS would be silently accepted here where real AWS returns InvalidArgument. It's genuinely untested too, every test in the diff happens to use compliant values (90/180), so nothing currently exercises the out-of-range case.

Also, the PR body doesn't follow the repo template (no Type of change, AWS Compatibility, or Checklist sections), worth using it since this repo squash-merges and the body becomes the commit message.

**Comment by ChiaraFor96** [comment]
> Nicely mirrors the existing metrics-configuration pattern, checked the request-shape validation (Filter exclusivity, AccessTier values, the concurrency handling for the delete/recreate race) against the real docs and it all matches. Good to see a genuine XXE test in there too, not just a happy-path check.
> 
> One real gap though: Days is only validated as a positive integer, but AWS's own Tiering reference is explicit that the minimum is tier-specific, 90 days for ARCHIVE_ACCESS, 180 for DEEP_ARCHIVE_ACCESS, capped at 730 for both. Right now Days=1 on ARCHIVE_ACCESS would be silently accepted here where real AWS returns InvalidArgument. It's genuinely untested too, every test in the diff happens to use compliant values (90/180), so nothing currently exercises the out-of-range case.
> 
> Also, the PR body doesn't follow the repo template (no Type of change, AWS Compatibility, or Checklist sections), worth using it since this repo squash-merges and the body becomes the commit message.

Add tests to cover the AWS's own Tiering reference and error. 
Updated also the PR body

File: src/main/java/io/github/hectorvent/floci/services/s3/S3Controller.java [hunk 3]
```diff
@@ -419,6 +428,15 @@ public Response deleteBucket(@PathParam("bucket") String bucket,
                 s3Service.deleteBucketMetricsConfiguration(bucket, requireMetricsId(uriInfo));
                 return Response.noContent().build();
             }
+            if (hasQueryParam(uriInfo, "intelligent-tiering")) {
+                // Likewise this must not fall through to deleting the bucket.
+                S3Service.RequestAuthorization authorization = S3RequestAuthorizationParser.parseIfRequired(
+                        s3Service.isAuthEnforced(), httpHeaders, uriInfo);
+                s3Service.authorizeBucketWrite(bucket, "s3:PutIntelligentTieringConfiguration", authorization);
```
[the remark this answers is not in the document]
Comment (ChiaraFor96) [reply src/main/java/io/github/hectorvent/floci/services/s3/S3Controller.java:435]: | DeleteBucketIntelligentTieringConfiguration | (Required) s3:PutIntelligentTieringConfiguration | Required to delete the existing S3 Intelligent-Tiering configuration from an S3 bucket. |

Source: [Required permissions for Amazon S3 API operations](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-with-s3-policy-actions.html)

This is a common AWS pattern. There is no separate s3:DeleteIntelligentTieringConfiguration action.

**Review by pgermosen — APPROVED** [review]
This looks good now. The template is filled in properly, and the Days validation is exactly what real S3 does: 90-day minimum for ARCHIVE_ACCESS, 180 for DEEP_ARCHIVE_ACCESS, capped at 730 for both, with ArgumentName/ArgumentValue in the error body, and there are direct tests pinning the boundaries.

Also good to see the null-body and Days-overflow edge cases tightened up so they come back as MalformedXML instead of a 500, and the auth check added on the PUT path so an unsigned request can't create or replace a configuration when enforcement is on. On the DELETE-permission question, checked it myself too: AWS really does reuse s3:PutIntelligentTieringConfiguration for the delete operation, there's no separate delete action, so that one's correct as written.

Nothing else outstanding. Nice work chasing all of this down.
